clientv3/integration/watch_test.go

```diff
@@ -396,3 +396,99 @@ func TestWatchCompactRevision(t *testing.T) {
 		t.Fatalf("expected closed channel, but got %v", wresp)
 	}
 }
+
+func TestWatchWithProgressNotify(t *testing.T)        { testWatchWithProgressNotify(t, true) }
+func TestWatchWithProgressNotifyNoEvent(t *testing.T) { testWatchWithProgressNotify(t, false) }
+func testWatchWithProgressNotify(t *testing.T, matchedEvent bool) {
```

Comment:
@gyuho @heyitsanthony 

I still feel this test is too complicated. 

I guess we should create a watcher and test case by case. Do not do any concurrent put. Do not mix receiving creation phase and receiving event message phase.